common/remote/sockfile.cpp

```diff
@@ -4020,20 +4171,216 @@ IRemoteActivity *createRemoteDiskRead(IPropertyTree &actNode)
+        return nullptr;
+    }
+    virtual void serializeCursor(MemoryBuffer &tgt) const override
+    {
+        keyManager->serializeCursorPos(tgt);
+        tgt.append(processed);
+    }
+    virtual void restoreCursor(MemoryBuffer &src) override
+    {
+        checkOpen();
+        eofSeen = false;
+        keyManager->deserializeCursorPos(src);
+        src.read(processed);
+    }
+    virtual StringBuffer &getInfoStr(StringBuffer &out) const override
+    {
+        return out.appendf("indexread[%s]", helper->getFileName());
+    }
+//interface IVirtualFieldCallback
+    virtual unsigned __int64 getFilePosition(const void * row) override
+    {
+        return 0; // JCSMORE
+    }
+    virtual unsigned __int64 getLocalFilePosition(const void * row) override
+    {
+        return 0; // JCSMORE
+    }
+};
+
+
+IRemoteActivity *createRemoteIndexRead(IPropertyTree &actNode)
+{
+    const char *fileName = actNode.queryProp("fileName");
+    unsigned __int64 chooseN = actNode.getPropInt64("chooseN", defaultFileStreamChooseNLimit);
+    unsigned __int64 skipN = actNode.getPropInt64("skipN", defaultFileStreamSkipN);
+    bool isTlk = actNode.getPropBool("isTlk");
+    bool allowPreload = actNode.getPropBool("allowPreload");
+
+    Owned<IOutputMetaData> inMeta = getTypeInfoOutputMetaData(actNode, "input", false);
+    Owned<IOutputMetaData> outMeta = getTypeInfoOutputMetaData(actNode, "output", false);
+    if (!outMeta)
+        outMeta.set(inMeta);
+    Owned<IHThorIndexReadArg> helper = createIndexReadArg(fileName, LINK(inMeta), LINK(outMeta), LINK(outMeta), chooseN, skipN, (unsigned __int64)-1);
+    assertex(!helper->needTransform()); // Code assumes that we do not need to call the transform
+    IDynamicIndexReadArg *dynArg = QUERYINTERFACE(helper.get(), IDynamicIndexReadArg);
+    assertex(dynArg);
+    Owned<IPropertyTreeIterator> filterIter = actNode.getElements("keyFilter");
+    ForEach(*filterIter)
+        dynArg->addFilter(filterIter->query().queryProp(nullptr));
+    return new CRemoteIndexReadActivity(helper, isTlk, allowPreload);
+}
+
+
+class CRemoteIndexCountActivity : public CRemoteIndexReadActivity
+{
+    typedef CRemoteIndexReadActivity PARENT;
+public:
+    CRemoteIndexCountActivity(IHThorIndexReadArg *_helper, bool _isTlk, bool _allowPreload) : PARENT(_helper, _isTlk, _allowPreload)
+    {
+    }
+// IRemoteActivity impl.
+    virtual const void *nextRow(MemoryBufferBuilder &outBuilder, size32_t &retSz) override
+    {
+        if (eofSeen)
+            return nullptr;
+        checkOpen();
+        unsigned __int64 count = 0;
+        if (!eofSeen)
+        {
+            if (helper->getRowLimit())
+                count = keyManager->checkCount(helper->getRowLimit());
+            else
+                count = keyManager->getCount();
+        }
+        void *tgt = outBuilder.ensureCapacity(sizeof(count), "count");
+        const void *ret = outBuilder.getSelf();
+        memcpy(tgt, &count, sizeof(count));
```

Comment:
does an index exists return 8 bytes or a single byte?  If a single then this might cause problems.